Problem with installation

While installing cutorch using luarocks I'm getting following error:

Warning: unmatched variable LUALIB

jopts=$(getconf _NPROCESSORS_CONF)

echo "Building on $jopts cores"
cmake -E make_directory build && cd build && cmake .. -DLUALIB= -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="/home/ubuntu/torch/install/bin/.." -DCMAKE_INSTALL_PREFIX="/home/ubuntu/torch/install/lib/luarocks/rocks/cutorch/scm-1" && make -j$jopts install

Building on 8 cores
-- Found Torch7 in /home/ubuntu/torch/install
-- MAGMA not found. Compiling without MAGMA support
-- Autodetected CUDA architecture(s): 3.0
-- got cuda version 7.0
-- Could not find CUDA with FP16 support, compiling without torch.CudaHalfTensor
-- CUDA_NVCC_FLAGS: -gencode;arch=compute_30,code=sm_30
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ubuntu/cutorch/build
[  3%] [  3%] [  3%] [  4%] [  6%] [  7%] Building NVCC (Device) object lib/THC/CMakeFiles/THC.dir//./THC_generated_THCTensorSort.cu.o
Building NVCC (Device) object lib/THC/CMakeFiles/THC.dir/generated/./THC_generated_THCTensorSortByte.cu.o
[  8%] Building NVCC (Device) object lib/THC/CMakeFiles/THC.dir/generated/./THC_generated_THCTensorMathCompareTByte.cu.o
Building NVCC (Device) object lib/THC/CMakeFiles/THC.dir//./THC_generated_THCTensorTypeUtils.cu.o
Building NVCC (Device) object lib/THC/CMakeFiles/THC.dir//./THC_generated_THCTensorRandom.cu.o
[  9%] Building NVCC (Device) object lib/THC/CMakeFiles/THC.dir/generated/./THC_generated_THCTensorMathPointwiseByte.cu.o
Building NVCC (Device) object lib/THC/CMakeFiles/THC.dir/generated/./THC_generated_THCTensorMathCompareByte.cu.o
Building NVCC (Device) object lib/THC/CMakeFiles/THC.dir/generated/./THC_generated_THCTensorMathReduceByte.cu.o
/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(221): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(221): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(221): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(225): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(225): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(225): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(229): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(229): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(229): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(233): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(233): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/THCTensorRandom.cu(233): error: identifier "half" is undefined

/home/ubuntu/cutorch/lib/THC/generic/THCTensorRandom.cu(17): error: more than one instance of overloaded function "generate_uniform" matches the argument list:
            function "generate_uniform(curandStateMtgp32 *, int, float *, double, double)"
            function "generate_uniform(curandStateMtgp32 *, int, <error-type> *, double, double)"
            argument types are: (curandStateMtgp32 *, ptrdiff_t, float *, double, double)

/home/ubuntu/cutorch/lib/THC/generic/THCTensorRandom.cu(31): error: more than one instance of overloaded function "generate_normal" matches the argument list:
            function "generate_normal(curandStateMtgp32 *, int, float *, double, double)"
            function "generate_normal(curandStateMtgp32 *, int, <error-type> *, double, double)"
            argument types are: (curandStateMtgp32 *, ptrdiff_t, float *, double, double)

/home/ubuntu/cutorch/lib/THC/generic/THCTensorRandom.cu(62): error: more than one instance of overloaded function "generate_exponential" matches the argument list:
            function "generate_exponential(curandStateMtgp32 *, int, float *, double)"
            function "generate_exponential(curandStateMtgp32 *, int, <error-type> *, double)"
            argument types are: (curandStateMtgp32 *, ptrdiff_t, float *, double)

/home/ubuntu/cutorch/lib/THC/generic/THCTensorRandom.cu(77): error: more than one instance of overloaded function "generate_cauchy" matches the argument list:
            function "generate_cauchy(curandStateMtgp32 *, int, float *, double, double)"
            function "generate_cauchy(curandStateMtgp32 *, int, <error-type> *, double, double)"
            argument types are: (curandStateMtgp32 *, ptrdiff_t, float *, double, double)

/home/ubuntu/cutorch/lib/THC/generic/THCTensorRandom.cu(17): error: more than one instance of overloaded function "generate_uniform" matches the argument list:
            function "generate_uniform(curandStateMtgp32 *, int, double *, double, double)"
            function "generate_uniform(curandStateMtgp32 *, int, <error-type> *, double, double)"
            argument types are: (curandStateMtgp32 *, ptrdiff_t, double *, double, double)

/home/ubuntu/cutorch/lib/THC/generic/THCTensorRandom.cu(31): error: more than one instance of overloaded function "generate_normal" matches the argument list:
            function "generate_normal(curandStateMtgp32 *, int, double *, double, double)"
            function "generate_normal(curandStateMtgp32 *, int, <error-type> *, double, double)"
            argument types are: (curandStateMtgp32 *, ptrdiff_t, double *, double, double)

/home/ubuntu/cutorch/lib/THC/generic/THCTensorRandom.cu(62): error: more than one instance of overloaded function "generate_exponential" matches the argument list:
            function "generate_exponential(curandStateMtgp32 *, int, double *, double)"
            function "generate_exponential(curandStateMtgp32 *, int, <error-type> *, double)"
            argument types are: (curandStateMtgp32 *, ptrdiff_t, double *, double)

/home/ubuntu/cutorch/lib/THC/generic/THCTensorRandom.cu(77): error: more than one instance of overloaded function "generate_cauchy" matches the argument list:
            function "generate_cauchy(curandStateMtgp32 *, int, double *, double, double)"
            function "generate_cauchy(curandStateMtgp32 *, int, <error-type> *, double, double)"
            argument types are: (curandStateMtgp32 *, ptrdiff_t, double *, double, double)

20 errors detected in the compilation of "/tmp/tmpxft_00003fd4_00000000-7_THCTensorRandom.cpp1.ii".
CMake Error at THC_generated_THCTensorRandom.cu.o.cmake:267 (message):
  Error generating file
  /home/ubuntu/cutorch/build/lib/THC/CMakeFiles/THC.dir//./THC_generated_THCTensorRandom.cu.o


make[2]: *** [lib/THC/CMakeFiles/THC.dir/./THC_generated_THCTensorRandom.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
^Cmake[2]: *** [lib/THC/CMakeFiles/THC.dir/generated/./THC_generated_THCTensorMathPointwiseByte.cu.o] Interrupt
make[2]: *** [lib/THC/CMakeFiles/THC.dir/generated/./THC_generated_THCTensorSortByte.cu.o] Interrupt
make[1]: *** [lib/THC/CMakeFiles/THC.dir/all] Interrupt
make: *** [all] Interrupt

Error: Build error: Failed building.

Please help.

--- a/lib/THC/THCTensorRandom.cu
+++ b/lib/THC/THCTensorRandom.cu
@@ -218,19 +218,22 @@ __global__ void NAME(curandStateMtgp32 *state, int size, T *result, ARG1, ARG2) 
 
 GENERATE_KERNEL2(generate_uniform, float, double a, double b, float, curand_uniform, x * (b-a) + a)
 GENERATE_KERNEL2(generate_uniform, double, double a, double b, double, curand_uniform_double, x * (b-a) + a)
-GENERATE_KERNEL2(generate_uniform, half, double a, double b, float, curand_uniform, (ScalarConvert<float, half>::to(x * (b-a) + a)))
 
 GENERATE_KERNEL2(generate_normal, float, double mean, double stdv, float, curand_normal, (x * stdv) + mean)
 GENERATE_KERNEL2(generate_normal, double, double mean, double stdv, double, curand_normal_double, (x * stdv) + mean)
-GENERATE_KERNEL2(generate_normal, half, double mean, double stdv, float, curand_normal, (ScalarConvert<float, half>::to((x * stdv) + mean)))
 
 GENERATE_KERNEL1(generate_exponential, float, double lambda, float, curand_uniform, (float)(-1. / lambda * log(1-x)))
 GENERATE_KERNEL1(generate_exponential, double, double lambda, double, curand_uniform_double, (double)(-1. / lambda * log(1-x)))
-GENERATE_KERNEL1(generate_exponential, half, double lambda, float, curand_uniform, (ScalarConvert<float, half>::to((float)(-1. / lambda * log(1-x)))))
 
 GENERATE_KERNEL2(generate_cauchy, float, double median, double sigma, float, curand_uniform, (float)(median + sigma * tan(M_PI*(x-0.5))))
 GENERATE_KERNEL2(generate_cauchy, double, double median, double sigma, double, curand_uniform_double, (double)(median + sigma * tan(M_PI*(x-0.5))))
+
+#ifdef CUDA_HALF_TENSOR
+GENERATE_KERNEL2(generate_uniform, half, double a, double b, float, curand_uniform, (ScalarConvert<float, half>::to(x * (b-a) + a)))
+GENERATE_KERNEL2(generate_normal, half, double mean, double stdv, float, curand_normal, (ScalarConvert<float, half>::to((x * stdv) + mean)))
+GENERATE_KERNEL1(generate_exponential, half, double lambda, float, curand_uniform, (ScalarConvert<float, half>::to((float)(-1. / lambda * log(1-x)))))
 GENERATE_KERNEL2(generate_cauchy, half, double median, double sigma, float, curand_uniform, (ScalarConvert<float, half>::to((float)(median + sigma * tan(M_PI*(x-0.5))))))
+#endif // CUDA_HALF_TENSOR
 
 #include "generic/THCTensorRandom.cu"
 #include "THCGenerateAllTypes.h"

